doc/ug/dv_methodology.md

```diff
@@ -299,65 +299,124 @@ Once the regression tool is developed, it will provide a way to enable these cap
 
-*  **Assert Coverage**: This metric measures which assertions, cover properties and sequences have been exercised in the course of the simulation.
+* **Assert Coverage**: This metric measures which assertions, cover properties, and sequences have been exercised in the course of the simulation.
   Note, an assertion precondition counts as a cover point.
-*  **Covergroup Coverage**: This metric (sometimes called "Testbench Coverage") measures covergroups during simulation.
-  These are usually coded in the testbench to check that the desired stimulus was generated properly, but can also be embedded in the RTL.
-
-Code coverage can reach 100% but it may not be sufficient to indicate whether all interesting combinations of scenarios were exercised (such as boundary conditions, concurrent scenarios and different CSR fields holding specific combinations of values resulting in the DUT being exercised in interesting ways).
-Hence, developing a detailed functional coverage model in conjunction with constrained random test sequences is key to high quality DV effort.
-
-Functional coverage can also reach 100% while the code coverage may not.
-This would indicate that the testplan is either incomplete or there are undocumented or unreachable features in the design.
-As such they can be addressed accordingly through either updating the design specification and augmenting the testplan / written tests, or by optimizing the design to remove unreachable logic if possible.
-There may be features that cannot be tested and cannot be removed from the design either.
-These would have to be analyzed and excluded from the coverage collection as a waiver.
-This is an exercise the DV and the designer typically perform together.
-This is discussed in more detail below.
+* **Cover group Coverage**:  Observes values on busses, registers, etc.
+  can verify that a specific value or a range of values was seen on a bus.
+  And that no illegal values were seen.
+  Could also be the intersection of two cover points.
+  One example could be observing the FIFO full signal and the FIFO write enable signal is asserted at the same time.
+  This is called a coverage cross and will give valuable information on whether this scenario was ever seen. 
+* **Cover Property coverage**: This observes events or series of events.
+  As an example think of the TL UL register bus used.
+  A cover property cover point could be the handshaking between valid and ready.
+
+
+### Why do we need coverage
+
+The common answer is to flush out bugs in the design.
+This is not accurate enough. Making sure there are no bugs in a design, while important it is not sufficient.
+One must also make sure the design work as intended.
+That is providing all the functionality specified in the design specification.
+So a more precise answer as to why we need coverage is to flush out flaws in the design.
+Flaws being both design bugs and deficiencies in the design with respect to the specification.
```

Comment:
I think this should be explained before we explain the two types of coverage above. You could move this (as a new para) to line  309 before "There are two key types of coverage metrics:". This section heading is not needed.

Reply:
fixed